scripts/build-ui.sh

```diff
@@ -5,30 +5,45 @@
 
 set -e
 
-if [ -z "$UI_COMMITISH" ]; then
-	echo "Must set UI_COMMITISH"; exit 1
-fi
-
 if [ -z "$UI_CLONE_DIR" ]; then
 	echo "Must set UI_CLONE_DIR"; exit 1
 fi
 
 if [ -z "$UI_VERSION_FILE" ]; then
-	echo "Must set UI_CLONE_DIR"; exit 1
+	echo "Must set UI_VERSION_FILE"; exit 1
+fi
+
+UI_EDITION=$(make --no-print-directory edition)
+
+if [ "$UI_EDITION" == "oss" ]; then
+  UI_REPO=https://github.com/hashicorp/boundary-ui
+  REPO_NAME=boundary-ui
+else
+  UI_VERSION_FILE="${UI_VERSION_FILE}_ent"
+  UI_REPO=https://github.com/hashicorp/boundary-ui-enterprise
+  REPO_NAME=boundary-ui-enterprise
 fi
 
+UI_CURRENT_COMMIT=$(head -n1 < "${UI_VERSION_FILE}" | cut -d' ' -f1)
+
+if [ -z "$UI_COMMITISH" ]; then
+  echo "==> Building default UI version from $UI_VERSION_FILE: $UI_CURRENT_COMMIT"
+  export UI_COMMITISH="$UI_CURRENT_COMMIT"
+else
+  echo "==> Building custom UI version $UI_COMMITISH"
+fi;
+
 if which gh &> /dev/null;  then
     echo "Found gh cli, attempting to download ui assets"
 
-    UI_EDITION=$(make --no-print-directory edition)
-    artifact_id=$(gh api repos/hashicorp/boundary-ui/actions/artifacts | \
+    artifact_id=$(gh api "repos/hashicorp/${REPO_NAME}/actions/artifacts" --paginate | \
```

Comment:
We were only pulling the most recent ~30 artifacts. I added `--paginate` to actually look for all of them in case we're looking for an artifact that's a bit older but this could be a bit slower. Let me know if you think this is unnecessary as github artifacts are only stored for ~90 days anyways.

Reply:
I think you just kill 2 birds with one stone!! this will solve [this jira ticket we had](https://hashicorp.atlassian.net/browse/ICU-6689) 😃